ipaserver/plugins/group.py

```diff
@@ -505,6 +505,9 @@ def pre_callback(self, ldap, dn, entry_attrs, *keys, **options):
             else:
                 old_entry_attrs['objectclass'].append('ipaexternalgroup')
                 entry_attrs['objectclass'] = old_entry_attrs['objectclass']
+            if 'gidnumber' in entry_attrs:
+                raise errors.MutuallyExclusiveError(reason=_(
+                    'gid cannot be set for external group'))
```

Comment:
I would replace the message with 'neither gid nor posix type can be set for external group'. Otherwise I find it a bit puzzling:
```
# ipa group-mod group --posix --external
ipa: ERROR: gid cannot be set for external group
```

Reply:
I think maybe "An external group cannot be POSIX" which should cover all the cases.